### PR TITLE
Recover on creating a new epoch but failing to re-add anyone

### PR DESCRIPTION
**Pull request by Powersource**
Based on https://github.com/ssbc/ssb-tribes2/pull/124

Progresses https://github.com/ssbc/ssb-tribes2/issues/123

### CODE REVIEW

File: lib/exclude.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ function reAddMembers(ssb, groupId, opts, cb) {
         pull.collect((err) => {
           // prettier-ignore
           if (err) return cb(clarify(err, "Couldn't re-add remaining members after exclusion"))
-          cb()
+          cb(null, missingMembers)
```
Comment (Powersource) [inline lib/exclude.js:42]: remove, just for testing

File: listeners.js [hunk 2]
```diff
@@ -213,9 +213,11 @@ module.exports = function startListeners(ssb, config, onError) {
               const randomTimeout = timeoutScale * timeoutRandom
 
               const timeoutId = setTimeout(() => {
-                reAddMembers(ssb, group.id, null, (err) => {
+                reAddMembers(ssb, group.id, null, (err, readded) => {
                   // prettier-ignore
                   if (err && !isClosed) return onError(clarify(err, 'Failed re-adding members to epoch that missed some'))
+
+                  console.log('recover readd', readded)
```
[the remark this answers is not in the document]
Comment (Powersource) [reply listeners.js]: just had to add a `pull.unique('id')` after the group list stream to avoid group updates, we just need the id

**Comment by Powersource** [comment]
Apparently didn't need any extra logic for this

File: test/exclude-members.test.js [hunk 3]
```diff
@@ -776,3 +776,129 @@ test('On exclusion, if we fail to re-add all people, someone else does that inst
...
+      recps: [groupId],
+    })
+  )
+
+  await run('replicated', replicate(alice, bob, carol))
+
+  await run('carol accepts group invite', carol.tribes2.acceptInvite(groupId))
+
+  const additionsFeed = await run(
+    'get number of additions before exclude',
```
Comment (mixmix) [inline test/exclude-members.test.js:826]:
```suggestion
    'get additions feed',
```

File: index.js [hunk 4]
```diff
@@ -273,6 +273,9 @@ module.exports = {
                 // prettier-ignore
                 if (err) return cb(clarify(err, "Couldn't post init msg on new epoch when excluding members"))
 
+                // prettier-ignore
+                if (opts?._reAddCrash) return cb(new Error('Intentional crash before re-adding members'))
```
Comment (mixmix) [inline index.js:277]: Love this pattern.

**Review by mixmix — APPROVED** [review]
BEAUTIFUL

:desert_island: :cocktail: :fish:

File: index.js (hunk 4 above)
Comment (Powersource) [reply index.js:277]: yeah me too, great idea from you!

**Pull request merged**